models/issue_assignees.go

```diff
@@ -0,0 +1,232 @@
+// Copyright 2018 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package models
+
+import (
+	"fmt"
+
+	"code.gitea.io/gitea/modules/log"
+	api "code.gitea.io/sdk/gitea"
+
+	"github.com/go-xorm/xorm"
+)
+
+// IssueAssignees saves all issue assignees
+type IssueAssignees struct {
+	ID         int64 `xorm:"pk autoincr"`
+	AssigneeID int64 `xorm:"INDEX"`
+	IssueID    int64 `xorm:"INDEX"`
+}
+
+// This loads all assignees of an issue
+func (issue *Issue) loadAssignees(e Engine) (err error) {
+	var assigneeIDs []IssueAssignees
+
+	err = e.Where("issue_id = ?", issue.ID).Find(&assigneeIDs)
+	if err != nil {
+		return
+	}
+
+	for _, assignee := range assigneeIDs {
+		user, err := getUserByID(e, assignee.AssigneeID)
+		if err != nil {
+			user = NewGhostUser()
+			if !IsErrUserNotExist(err) {
+				return err
+			}
+		}
+		issue.Assignees = append(issue.Assignees, user)
+	}
+
+	// Check if we have at least one assignee and if yes put it in as `Assignee`
+	if len(issue.Assignees) > 0 {
+		issue.Assignee = issue.Assignees[0]
+	}
+
+	return
+}
+
+// GetAssigneesByIssue returns everyone assigned to that issue
+func GetAssigneesByIssue(issue *Issue) (assignees []*User, err error) {
+	err = issue.loadAssignees(x)
+	if err != nil {
+		return assignees, err
+	}
+
+	return issue.Assignees, nil
+}
+
+// IsUserAssignedToIssue returns true when the user is assigned to the issue
+func IsUserAssignedToIssue(issue *Issue, user *User) (isAssigned bool, err error) {
+	assignees := IssueAssignees{AssigneeID: user.ID, IssueID: issue.ID}
+	isAssigned, err = x.Get(&assignees)
```

Comment:
Use `x.Exist(&IssueAssignees{AssigneeID: user.ID, IssueID: issue.ID})` is better.

Reply:
I always forget this function exists :sweat_smile: